packages/ui-text-area/src/TextArea/styles.js

```diff
@@ -0,0 +1,123 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import generateComponentTheme from './theme'
+
+/**
+ * Generates the style object from the theme and provided additional information
+ * @param  {Object} theme The actual theme object.
+ * @param  {Object} themeOverride User provided overrides of the default theme mapping.
+ * @param  {Object} props the props of the component, the style is applied to
+ * @param  {Object} state the state of the component, the style is applied to
+ * @return {Object} The final style object, which will be used in the component
+ */
+const generateStyle = (theme, themeOverride, props, state) => {
+  // get the theme variables object for the component
+  const componentTheme = generateComponentTheme(theme, themeOverride)
+  const { disabled, size } = props
+
+  const fontSizeVariants = {
+    small: {
+      fontSize: componentTheme.smallFontSize
+    },
+    medium: {
+      fontSize: componentTheme.mediumFontSize
+    },
+    large: {
+      fontSize: componentTheme.largeFontSize
+    }
+  }
+
+  return {
+    textArea: {
+      label: 'textArea',
+      all: 'initial',
+      WebkitFontSmoothing: 'antialiased',
+      MozOsxFontSmoothing: 'grayscale',
+      appearance: 'none',
+      height: 'auto',
+      margin: 0,
+      width: '100%',
+      display: 'block',
+      boxSizing: 'border-box',
+      padding: componentTheme.padding,
+      fontFamily: componentTheme.fontFamily,
+      fontWeight: componentTheme.fontWeight,
+      borderWidth: componentTheme.borderWidth,
+      borderStyle: componentTheme.borderStyle,
+      borderTopColor: componentTheme.borderTopColor,
+      borderRightColor: componentTheme.borderRightColor,
+      borderBottomColor: componentTheme.borderBottomColor,
+      borderLeftColor: componentTheme.borderLeftColor,
+      borderRadius: componentTheme.borderRadius,
+      color: componentTheme.color,
+      background: componentTheme.background,
+      whiteSpace: 'pre-wrap',
+      wordWrap: 'break-word',
+      textAlign: 'start',
+      ...fontSizeVariants[size],
+      '&:focus + span': {
```

Comment:
Hm, i think using "span" here is pretty risky, the selector is too general. I would rather use the "[class$=-textArea__outline]" syntax, that is specific. Also, the css file used `~` in the selector, which is not equivalent to `+`.  But to be honest, the '+' should work here just as well...

Reply:
I've pushed an update that adresses this